mlflow/sagemaker/__init__.py

```diff
@@ -212,6 +222,47 @@ def _make_tarfile(output_filename, source_dir):
         for f in os.listdir(source_dir):
             tar.add(os.path.join(source_dir, f), arcname=f)
 
+def _get_account_id():
+    sess = boto3.Session()
+    sts_client = sess.client("sts")
+    identity_info = sts_client.get_caller_identity()
+    account_id = identity_info["Account"]
+    return account_id 
+
+def _get_assumed_role_arn():
+    """
+    :return: ARN of the user's current IAM role 
+    """
+    sess = boto3.Session()
+    sts_client = sess.client("sts")
+    identity_info = sts_client.get_caller_identity()
+    sts_arn = identity_info["Arn"]
+    role_name = sts_arn.split("/")[1]
+    iam_client = sess.client("iam")
+    role_response = iam_client.get_role(RoleName=role_name)
+    return role_response["Role"]["Arn"]
+
+def _get_default_s3_bucket(region_name):
+    # create bucket if it does not exist
+    account_id = _get_account_id()
+    bucket_name = "{pfx}-{aid}".format(pfx=DEFAULT_BUCKET_NAME_PREFIX, aid=account_id)
```

Comment:
btw, is the idea here that multiple accounts could access the same s3 bucket and so we want to distinguish them by account name? what bad things would happen if we just let the bucket name be the default name? what kind of naming convention does sagemaker itself use (when it creates the bucket for you)?

Reply:
S3 bucket names are global, so we're forced to create a unique bucket name. Come to think of it, this naming scheme doesn't handle the case where that bucket name is taken by a different account, which would be unfortunate.

Reply:
I'm not familiar with the sagemaker default bucket naming scheme. That should be explored

Reply:
From the sagemaker docs: ```When you make an Amazon SageMaker API call that accesses an S3 bucket location and one is not specified, the Session creates a default bucket based on a naming convention which includes the current AWS account ID.```

Reply:
ah we used to have some sagemaker-auto-generated S3 buckets in Models (e.g. your tensorflow one) which showed the naming convention. can't quite find them right now...